src/pushsource/_impl/backend/registry_source.py

```diff
@@ -90,36 +90,44 @@ def _push_item_from_registry_uri(self, uri, signing_key):
         else:
             klass = ContainerImagePushItem
 
-        if source_uri not in self._manifests:
-            manifest_details = get_manifest(
-                "%s://%s" % (schema, host),
-                repo,
-                src_tag,
-                manifest_types=[MT_S2_LIST],
-            )
-            self._manifests[source_uri] = manifest_details
+        for mtype in [MT_S2_V1, MT_S2_V2, MT_S2_LIST]:
+            if source_uri not in self._manifests:
+                self._manifests[source_uri] = {}
+            if mtype not in self._manifests[source_uri]:
+                manifest_details = get_manifest(
+                    "%s://%s" % (schema, host),
+                    repo,
+                    src_tag,
+                    manifest_types=[mtype],
+                )
+                self._manifests[source_uri][mtype] = manifest_details
 
-        manifest_details = self._manifests[source_uri]
-        content_type, _, _ = manifest_details
-        if content_type not in [MT_S2_V2, MT_S2_V1, MT_S2_V1_SIGNED, MT_S2_LIST]:
-            raise ValueError("Unsupported manifest type:%s" % content_type)
+            manifest_details = self._manifests[source_uri][mtype]
+            content_type, _, _ = manifest_details
+            if content_type not in [MT_S2_V2, MT_S2_V1, MT_S2_V1_SIGNED, MT_S2_LIST]:
```

Comment:
Doing this check for application/vnd.docker.distribution.manifest.v1+json breaks support for registry.access.redhat.com:

```
$ examples/list-push-items registry:image=registry.access.redhat.com/ubi8/ubi:8.4-211
# Loaded source registry:image=registry.access.redhat.com/ubi8/ubi:8.4-211
Traceback (most recent call last):
  File "examples/list-push-items", line 97, in <module>
    main()
  File "examples/list-push-items", line 93, in main
    run(p)
  File "examples/list-push-items", line 61, in run
    for pushitem in source:
  File "/home/rmcgover/src/pushsource/src/pushsource/_impl/backend/registry_source.py", line 151, in __iter__
    yield self._push_item_from_registry_uri(uri, key)
  File "/home/rmcgover/src/pushsource/src/pushsource/_impl/backend/registry_source.py", line 108, in _push_item_from_registry_uri
    raise ValueError("Unsupported manifest type:%s" % content_type)
ValueError: Unsupported manifest type:text/plain
```

You may recall that the CDN claims text/plain for the older types of manifest:

```
$ curl -I -L -H 'Accept: application/vnd.docker.distribution.manifest.v1+json' https://registry.access.redhat.com/v2/ubi8/ubi/manifests/8.4-211
HTTP/1.1 302 Moved Temporarily
Content-Length: 485
Content-Type: text/html; charset=utf-8
Location: https://registry.access.redhat.com/webassets/docker/content/dist/containers/rhel8/multiarch/appstream/containers/redhat-ubi8-ubi/manifests/1/8.4-211
Server: Apache/2.4.6 (Red Hat Enterprise Linux) OpenSSL/1.0.2k-fips mod_wsgi/3.4 Python/2.7.5
Expires: Mon, 29 Nov 2021 22:47:26 GMT
Cache-Control: max-age=0, no-cache, no-store
Pragma: no-cache
Date: Mon, 29 Nov 2021 22:47:26 GMT
Connection: keep-alive

HTTP/1.1 200 OK
Accept-Ranges: bytes
Content-Type: text/plain
ETag: "9382e58b99c34020b07ee0c9c1344ba4:1632219371.670031"
Last-Modified: Tue, 21 Sep 2021 10:06:30 GMT
Server: AkamaiNetStorage
Content-Length: 5778
Date: Mon, 29 Nov 2021 22:47:26 GMT
Connection: keep-alive
X-Docker-Size: -1
```

Can you please make it tolerate that case?

Also when you fix this, please don't propagate the wrong text/plain type up through the push item's media_type attribute (at `media_type=manifest[0]`) but instead fix it to application/vnd.docker.distribution.manifest.v1+json.  That will keep the quirk contained here rather than passing it on to users of the library.

Reply:
ah, yeah, I remember it. I thought we fixed that. Ok, I'll make code more tolerant